src/main/kotlin/com/example/fhirvalidator/configuration/ValidationConfiguration.kt

```diff
@@ -6,49 +6,75 @@ import ca.uhn.fhir.context.support.DefaultProfileValidationSupport
 import ca.uhn.fhir.context.support.IValidationSupport
 import ca.uhn.fhir.context.support.ValidationSupportContext
 import ca.uhn.fhir.validation.FhirValidator
+import com.example.fhirvalidator.model.ValidationConfig
 import com.example.fhirvalidator.service.ImplementationGuideParser
+import com.example.fhirvalidator.shared.AuthorisationClient
+import com.example.fhirvalidator.shared.HybridTerminologyValidationSupport
 import mu.KLogging
-import org.hl7.fhir.common.hapi.validation.support.CommonCodeSystemsTerminologyService
-import org.hl7.fhir.common.hapi.validation.support.InMemoryTerminologyServerValidationSupport
-import org.hl7.fhir.common.hapi.validation.support.SnapshotGeneratingValidationSupport
-import org.hl7.fhir.common.hapi.validation.support.ValidationSupportChain
+import org.hl7.fhir.common.hapi.validation.support.*
 import org.hl7.fhir.common.hapi.validation.validator.FhirInstanceValidator
 import org.hl7.fhir.instance.model.api.IBaseResource
-import org.hl7.fhir.r4.model.StructureDefinition
+import org.hl7.fhir.r4.model.*
 import org.hl7.fhir.utilities.npm.NpmPackage
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
 
+
 @Configuration
 class ValidationConfiguration(private val implementationGuideParser: ImplementationGuideParser) {
     companion object : KLogging()
 
+
     @Bean
     fun validator(fhirContext: FhirContext, instanceValidator: FhirInstanceValidator): FhirValidator {
-        return fhirContext.newValidator().registerValidatorModule(instanceValidator)
+        val validator = fhirContext.newValidator().registerValidatorModule(instanceValidator)
+        logger.info("Initialising Validator")
+        validator.validateWithResult(getInitialisationExampleResource())
+        return validator
     }
 
     @Bean
     fun instanceValidator(supportChain: ValidationSupportChain): FhirInstanceValidator {
-        return FhirInstanceValidator(supportChain)
+        return FhirInstanceValidator(CachingValidationSupport(supportChain))
     }
 
     @Bean
     fun validationSupportChain(
         fhirContext: FhirContext,
         terminologyValidationSupport: InMemoryTerminologyServerValidationSupport,
-        npmPackages: List<NpmPackage>
+        npmPackages: List<NpmPackage>,
+        validationConfig : ValidationConfig
     ): ValidationSupportChain {
         val supportChain = ValidationSupportChain(
-            DefaultProfileValidationSupport(fhirContext),
-            CommonCodeSystemsTerminologyService(fhirContext),
-            terminologyValidationSupport,
-            SnapshotGeneratingValidationSupport(fhirContext)
+            DefaultProfileValidationSupport(fhirContext)
         )
+        supportChain.addValidationSupport(CommonCodeSystemsTerminologyService(fhirContext))
+        if (validationConfig.useRemoteTerminology && !validationConfig.terminologyServer.isEmpty()) {
```

Comment:
Agreed. Suggest we push this into a later (terminology server) change.

Reply:
I'm planning to use just one bean for terminology support. This is a temporary change to test the new version.

Reply:
Possibly we could reuse the existing bean which uses InMemoryTerminologyServerValidationSupport

Move both code paths to use the single HybridTerminologyValidationSupport? (I'd left it separate to avoid changing existing code for EPS)

Some other notes here https://chat.fhir.org/#narrow/stream/179167-hapi/topic/Hybrid.20InMemoryTerminologyServerValidationSupport.20with.20Remot

Solutions Assurance are also looking at this class (that's why it is java not kotlin and potentially we could contribute this to HAPI FHIR)

Reply:
Moved to code to a bean.